### PR TITLE
fix non-thread-safe operation

### LINKED ISSUE #76: Seems after upadting to HA Core 2024.5.0 , call service on Hitachi Climate Entity will bring up error message
### 🐛 Describe the bug

Seems after upadting to HA Core 2024.5.0 , call service on Hitachi Climate Entity will bring up error message . (didn't encounter this problem before , until upadting to HA Core 2024.5.0 this moring ) .

error messgae as below:

服務 climate/set_temperature 執行失敗。 Detected that custom integration 'jcihitachi_tw' calls async_write_ha_state from a thread at custom_components/jcihitachi_tw/__init__.py, line 246: self.coordinator.async_set_updated_data(None). Please report it to the author of the 'jcihitachi_tw' custom integration.

### Environment Information

screenshoot

Core : 2024.5.0
Supervisor : 2024.04.4
Operating System : 12.2
Device : RaspberryPi 4b

### PR DESCRIPTION
**Pull request by cloudrainstar**
I followed the fix mentioned here: https://github.com/home-assistant/core/issues/115235#issuecomment-2043832269 and adapted it here, and it looks like it should fix #76.

### CODE REVIEW

File: custom_components/jcihitachi_tw/__init__.py [hunk 1]
```diff
@@ -24,6 +24,10 @@
 BASE_TIMEOUT = 5
 
 
+class JciHitachiDataUpdateCoordinator(DataUpdateCoordinator):
+    def set_updated_data(self, device=None) -> None:
+        self.hass.loop.call_soon_threadsafe(self.async_set_updated_data, device)
```
Comment (qqaatw) [inline custom_components/jcihitachi_tw/__init__.py:29]: Hmm, `device` looks to me unneeded no?

**Review by qqaatw — COMMENTED** [review]
Have you tested this locally with Home Assistant core `2024.5` onward?

**Comment by cloudrainstar** [comment]
> Have you tested this locally with Home Assistant core `2024.5` onward?

Yes, I updated my HA to 2024.5.1 yesterday and ran into this issue, so I found other custom components had the same issue and found that their fix solves the issue here. I've been running my HA instance on this code change locally since yesterday.

File: custom_components/jcihitachi_tw/__init__.py (hunk 1 above)
Comment (cloudrainstar) [reply custom_components/jcihitachi_tw/__init__.py:29]: Yes, for this implementation we can default to None and it should be ok. The `device` is there to be consistent with the original `async_set_updated_data` function. I can remove it if you prefer it that way.

File: custom_components/jcihitachi_tw/__init__.py [hunk 2]
```diff
@@ -243,4 +247,4 @@ def update(self):
         )
         
         # Important: We have to reset the update scheduler to prevent old status from wrongly being loaded. 
-        self.coordinator.async_set_updated_data(None)
+        self.coordinator.set_updated_data(None)
```
Comment (qqaatw) [inline custom_components/jcihitachi_tw/__init__.py:250]:
```suggestion
        self.hass.loop.call_soon_threadsafe(self.coordinator.async_set_updated_data, None)
```

Let's discard the rest of change and update this line. Thanks for your contribution.

File: custom_components/jcihitachi_tw/__init__.py (hunk 1 above)
Comment (qqaatw) [reply custom_components/jcihitachi_tw/__init__.py:29]: See below, thanks.

**Review by qqaatw — APPROVED** [review]
Cheers

**Pull request merged**